ui/plugins/auto_classification/controller.js

```diff
@@ -1,22 +1,745 @@
+                    // new entry
+                    best = new ThClassificationOption("classified_failure",
+                                                      bestClassification,
+                                                      null,
+                                                      "",
+                                                      getClassificationMatches(bestClassification));
+                }
+                ui.options = [best].concat(ui.options);
+            }
+            return best;
+        }
+
+        function buildUIData(data, matchers) {
+            var ui = {
+                options: [],
+                best: null,
+            };
+
+            var getClassificationMatches = getClassifiedFailureMatcher(matchers,
+                                                                       data.matches);
+
+            var autoOptions = autoclassifierOptions(data.classified_failures,
+                                                    getClassificationMatches);
+
+            var bugSuggestions = filterUnstructuredBugs(data.unstructured_bugs, autoOptions);
+            ui.options = ui.options.concat(autoOptions,
+                                           bugSuggestionOptions(bugSuggestions));
+            ui.best = bestAutoclassifiedOption(data.best_classification, ui,
+                                               getClassificationMatches);
+
+            ui.options.push(new ThClassificationOption("manual", "manual"));
+
+            var ignoreOption = new ThClassificationOption("ignore", "ignore", 0);
+            ui.options.push(ignoreOption);
+            if (ui.best === "ignore") {
+                ui.best = ignoreOption;
+                ignoreOption.always = true;
+            } else if(ui.best && !ui.best.hasBug) {
+                // If we have a best option with no bug, the UI won't display a
+                // radio option for that case, set the default to the following option
+                ui.selectedOptionIndex = 1;
+            }
+            if (ui.best) {
+                ui.best.isBest = true;
+            }
+
+            return ui;
+        }
+
+        var ThStructuredLine = function(data, matchers) {
+            var lineInterface = {
+                type: "structured",
+                data: data, // This rather breaks the interface but it makes things a bit easier for now
+                options: null,
+                best: null,
+                selectedOptionIndex: 0,
+                dirty: false, // Has line changed from its default value
+                updateSelectOption: "Update",
+
+                get id() {
+                    return this.data.id;
+                },
+
+                get status() {
+                    if (data.best_is_verified) {
+                        if (data.best_classification === null ||
+                            data.best_classification.bug_number === 0) {
+                            var rv = 'ignored';
```

Comment:
``var rv`` should be declared at an outer scope.  Maybe up above line 454.  Otherwise, 2 lines later, rv was never declared.